invenio_rdm_records/services/config.py

```diff
@@ -295,6 +338,36 @@ class RDMFileRecordServiceConfig(FileServiceConfig, ConfiguratorMixin):
     file_schema = FileSchema
 
 
+class RDMThumbnailLink(RecordLink):
+    """RDM thumbnail links dictionary."""
+
+    def __init__(self, *args, sizes=None, **kwargs):
+        """Constructor."""
+        self._sizes = sizes
+        super().__init__(*args, **kwargs)
+
+    def expand(self, obj, context):
+        """Expand the thumbs size dictionary of URIs."""
+        vars = {}
+        vars.update(deepcopy(context))
+        self.vars(obj, vars)
+        if self._vars_func:
+            self._vars_func(obj, vars)
+        vars = preprocess_vars(vars)
+
+        thumbnail_links = {}
+        for size in self._sizes():
+            try:
+                thumbnail_url = record_thumbnail(size, record=obj)
+                thumbnail_links[str(size)] = (
+                    self._uritemplate.expand(**vars) + thumbnail_url
+                )
```

Comment:
```suggestion
        thumbnail_links = {}
        vars["file_key"] = get_record_thumbnail_file(record=obj)
        for size in self._sizes:
            thumbnail_links[str(size)] = self._uritemplate.expand(**vars)
```

- We can compute the thumbnail file once outside the loop, since the file will remain the same for all sizes
- No need for try/except anymore